### PR TITLE
events: provide more details when operator status change

### PR DESCRIPTION
**Pull request by mfojtik**
This change provides more details into what changed in operator status conditions, instead of just printing the status changed.

/cc @deads2k 
/cc @smarterclayton

### CODE REVIEW

File: pkg/config/clusteroperator/v1helpers/status.go [hunk 1]
```diff
@@ -55,6 +57,32 @@ func FindStatusCondition(conditions []configv1.ClusterOperatorStatusCondition, c
...
+func GetStatusConditionDiff(oldConditions []configv1.ClusterOperatorStatusCondition, newConditions []configv1.ClusterOperatorStatusCondition) string {
+	messages := []string{}
+	for _, newCondition := range newConditions {
+		existingStatusCondition := FindStatusCondition(oldConditions, newCondition.Type)
+		if existingStatusCondition == nil {
+			messages = append(messages, fmt.Sprintf("%s set to %s (%s)", newCondition.Type, newCondition.Status, newCondition.Message))
+			continue
+		}
+		if existingStatusCondition.Status != newCondition.Status {
+			messages = append(messages, fmt.Sprintf("%s changed from %s to %s (%s)", existingStatusCondition.Type, existingStatusCondition.Status, newCondition.Status, newCondition.Message))
```
Comment (sttts) [inline pkg/config/clusteroperator/v1helpers/status.go]: `%q` for the message

Comment (sttts) [reply pkg/config/clusteroperator/v1helpers/status.go]: same above

Comment (sttts) [reply pkg/config/clusteroperator/v1helpers/status.go]: `%s status changed ...`

Comment (mfojtik) [reply pkg/config/clusteroperator/v1helpers/status.go]: yes, fixed.

**Pull request merged**